src/routes/InputCardPage.tsx

```diff
@@ -44,8 +45,11 @@ export default function InputCardPage() {
 
   React.useEffect(() => {
     setHideCancelButton(
-      !!(getSessionItem(SessionItems.transaction) as Transaction | undefined)
-        ?.transactionId
+      !!(
+        getSessionItem(SessionItems.transaction) as
+          | NewTransactionResponse
```

Comment:
In this case you can use fp-ts  

```suggestion
      !!(
```
        pipe(
          getSessionItem(SessionItems.transaction),
          NewTransactionResponse.decode,
          E.fold(
            () => undefined,
            (transaction) => transaction.transactionId
          )
        )
      )
    );

Reply:
done!